fe/fe-core/src/main/java/com/starrocks/lake/Utils.java

```diff
@@ -204,4 +206,21 @@ public static void publishLogVersionBatch(@NotNull List<Tablet> tablets, List<Lo
             }
         }
     }
+
+    public static long getFirstWorkerGroupByWarehouseId(WarehouseManager manager, long warehouseId) {
+        Warehouse warehouse = manager.getWarehouse(warehouseId);
+        List<Long> ids = warehouse.getWorkerGroupIds();
+        return ids.get(0);
+    }
+
+    public static long getWarehouseIdByBackendId(SystemInfoService systemInfo, long nodeId) {
```

Comment:
Optional<Long>, let the caller determine what to use.

Reply:
I do not get your point.

Reply:
`Optional<Long> get....`

It could be possible that no warehouse can be found by the nodeId, let the caller decide which warehouse to use then.

The caller should do something like
```
long warehouseId = getWarehouseIdByBackendId(systemInfo, nodeId).orElse(WarehouseManager.DEFAULT_WAREHOUSE_ID);
```